tcp/src/stream.cc

```diff
@@ -19,6 +19,7 @@
 #include <QWaitCondition>
 #include <sstream>
 #include <sys/socket.h>
+#include <sys/timer.h>
```

Comment:
Could you please explain why this header is needed ?

Reply:
I made a small change to check if `timer.h` exists, because I tested it now on debian8 and I had a fail in that.
Ref: https://www.gnu.org/software/autoconf/manual/autoconf-2.67/html_node/Particular-Headers.html

Reply:
OK the inclusion of the header is explained. However the proper header would be <sys/time.h> as far as POSIX is concerned. No need to check it for existence.

Once this is fixed your PR could be merged.